src/pdf/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,236 @@
+import {
+    fetchClients,
+    fetchFamilies,
+    fetchLists,
+    fetchParcels,
+} from "@/pdf/ShoppingList/databaseFetch";
+import { Schema } from "@/supabase";
+import { Person } from "@/components/Form/formFunctions";
+
+export interface BlockProps {
+    [key: string]: string;
+}
+
+interface ParcelInfoAndClientID {
+    parcelInfo: ParcelInfo;
+    clientID: string;
+}
+
+interface ClientDataAndFamilyData {
+    clientData: Schema["clients"];
+    familyData: Schema["families"][];
+}
+
+interface NappySizeAndExtraInformation {
+    nappySize: string;
+    extraInformation: string;
+}
+
+export interface ShoppingListPDFProps {
+    postcode: string;
+    parcelInfo: ParcelInfo;
+    clientSummary: ClientSummary;
+    householdSummary: HouseholdSummary;
+    requirementSummary: RequirementSummary;
+    itemsList: Item[];
+    endNotes: string;
+}
+
+export interface ParcelInfo extends BlockProps {
+    voucherNumber: string;
+    packingDate: string;
+    collectionDate: string;
+    collectionSite: string;
+}
+
+export interface ClientSummary {
+    name: string;
+    contact: string;
+    address: string;
+    extraInformation: string;
+}
+
+export interface HouseholdSummary extends BlockProps {
+    householdSize: string;
+    genderBreakdown: string;
+    numberOfBabies: string;
+    ageAndGenderOfChildren: string;
+}
+
+export interface RequirementSummary extends BlockProps {
+    feminineProductsRequired: string;
+    babyProductsRequired: string;
+    petFoodRequired: string;
+    dietaryRequirements: string;
+    otherItems: string;
+}
+
+export interface Item {
+    description: string;
+    quantity: string;
+    notes: string;
+}
+
+const formatDate = (dateString: string | null): string => {
+    if (dateString === null) {
```

Comment:
is it worth changing this to `if (!dateString)` instead to protect against an empty string as a date?